### PR TITLE
chore: 🤖 upgrade electron

### PR DESCRIPTION
**Pull request by cameronperera**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-13697

# Description

I have updated electron to v31. I've gone through the electron breaking [changes](https://www.electronjs.org/docs/latest/breaking-changes) and fixed one breaking change that affected us.  

## How to Test
[Here's](https://github.com/hashicorp/boundary-ui-releases/actions/runs/9570586017) a build in our CI of this electron upgrade.

Confirm that the app runs. Note that this includes the client-daemon to cache-daemon changes which the currently CLI hasn't updated yet so you'll see errors for that.

## Checklist

- ~[ ] I have added before and after screenshots for UI changes~
- ~[ ] I have added JSON response output for API changes~
- ~[ ] I have added steps to reproduce and test for bug fixes in the description~
- ~[ ] I have commented on my code, particularly in hard-to-understand areas~
- [x] My changes generate no new warnings
- ~[ ] I have added tests that prove my fix is effective or that my feature works~

### CODE REVIEW

File: ui/desktop/electron-app/package.json [hunk 1]
```diff
@@ -40,10 +40,11 @@
...
     "@electron-forge/maker-squirrel": "^7.2.0",
     "@electron-forge/maker-zip": "^7.2.0",
-    "electron": "26.2.4",
+    "electron": "28.3.3",
     "unzipper": "^0.11.5"
   },
   "resolutions": {
-    "highlight.js": "^10.4.1"
+    "highlight.js": "^10.4.1",
+    "node-abi": "3.64.0"
```
Comment (ZedLi) [inline ui/desktop/electron-app/package.json:48]: You shouldn't need a resolution here it looks like. Our `node-abi` versions can be brought up manually to 3.64.0 in the `yarn.lock` file without a resolution.

Comment (cameronperera) [reply ui/desktop/electron-app/package.json:48]: Right we have chatted about this before a while back. I will update this.

Comment (cameronperera) [reply ui/desktop/electron-app/package.json]: Updated. Let me know if I updated the correct parts.

**Review by ZedLi — APPROVED** [review]

**Comment by ZedLi** [comment]
> When trying to upgrade to v29 and v30, I started running into node-gyp build issues.

I took a look into this, if you also upgrade `nan`, it should also fix the issue.

**Review by ZedLi — COMMENTED** [review]
The `crashed` event was [removed](https://www.electronjs.org/docs/latest/breaking-changes#removed-crashed-event-on-webcontents-and-webview) which we'll have to update

File: ui/desktop/electron-app/src/index.js [hunk 2]
```diff
@@ -125,7 +125,7 @@ const createWindow = (partition, closeWindowCB) => {
     browserWindow.loadURL(emberAppURL);
   });
 
-  browserWindow.webContents.on('crashed', () => {
+  browserWindow.webContents.on('render-process-gone', () => {
```
Comment (cameronperera) [inline ui/desktop/electron-app/src/index.js:128]: thanks for catching this change

**Review by calcaide — APPROVED** [review]

**Pull request merged**